services/_lambda.py

```diff
@@ -1,29 +1,95 @@
 from models import RuleCheckResult
 import boto3
+import json
 
 
-# client = boto3.client("")
+client = boto3.client("lambda")
+iam_client = boto3.client("iam")
 
 
 def lambda_dlq_check():
+    compliant_resource = []
+    non_compliant_resources = []
+    functions = client.list_functions()["Functions"]
+
+    for function in functions:
+        response = client.get_function(FunctionName=function["FunctionName"])["Configuration"]
+        if "DeadLetterConfig" in response:
+            compliant_resource.append(function["FunctionArn"])
+        else:
+            non_compliant_resources.append(function["FunctionArn"])
+
     return RuleCheckResult(
-        passed=False, compliant_resources=[], non_compliant_resources=[]
+        passed=not non_compliant_resources,
+        compliant_resources=compliant_resource,
+        non_compliant_resources=non_compliant_resources,
     )
 
 
 def lambda_function_public_access_prohibited():
+    compliant_resource = []
+    non_compliant_resources = []
+    functions = client.list_functions()["Functions"]
+
+    for function in functions:
+        try:
+            policy = json.loads(client.get_policy(FunctionName=function["FunctionName"])["Policy"])
+            for statement in policy["Statement"]:
+                if statement["Principal"] in ["*", "", '{"AWS": ""}', '{"AWS": "*"}']:
+                    non_compliant_resources.append(function["FunctionArn"])
+                    break
+            else:
+                compliant_resource.append(function["FunctionArn"])
+        except Exception as e:
+            if e.__class__.__name__ == "ResourceNotFoundException":
+                non_compliant_resources.append(function["FunctionArn"])
+            else:
+                raise e
+
     return RuleCheckResult(
-        passed=False, compliant_resources=[], non_compliant_resources=[]
+        passed=not non_compliant_resources,
+        compliant_resources=compliant_resource,
+        non_compliant_resources=non_compliant_resources,
     )
 
 
 def lambda_function_settings_check():
+    compliant_resource = []
+    non_compliant_resources = []
+    functions = client.list_functions()["Functions"]
+
+    runtime = []  # python3.7 | nodejs10.x ...
+
+    for function in functions:
+        configuration = client.get_function(FunctionName=function["FunctionName"])["Configuration"]
+
+        if configuration["Runtime"] in runtime:
+            compliant_resource.append(function["FunctionArn"])
+        else:
+            non_compliant_resources.append(function["FunctionArn"])
```

Comment:
기존의 config rule이 rule에 설정한 값과 function의 설정 값을 비교하는 것인데, 이 부분은 필요에 맞게 수정해서 사용하겠습니다.
```suggestion
    default_timeout = 3
    default_memory_size = 128

    for function in functions:
        if function["Timeout"] == default_timeout or function["MemorySize"] == default_memory_size:
            non_compliant_resources.append(function["FunctionArn"])
        else:
            compliant_resource.append(function["FunctionArn"])
```